### PR TITLE
Delete `OBJ(LCL_ADDR)` args handling from local morph ref counting

### PR DESCRIPTION
**Pull request by SingleAccretion**
No longer needed.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
[DIiffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=256735&view=ms.vss-build-web.run-extensions-tab).

There is one regression, but the TP diffs should outweigh that by a lot: `0.3%` - `0.4%` wins in MinOpts, `0.2%`-`0.3%` in FullOpts.

@dotnet/jit-contrib

File: src/coreclr/jit/lclmorph.cpp [hunk 1]
```diff
@@ -1535,14 +1534,16 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
...
+    //    user   - "node"'s user
     //
     // Notes:
     //    fgMakeOutgoingStructArgCopy checks the ref counts for implicit byref params when it decides
     //    if it's legal to elide certain copies of them;
     //    fgRetypeImplicitByRefArgs checks the ref counts when it decides to undo promotions.
     //    fgForwardSub uses ref counts to decide when to forward sub.
     //
-    void UpdateEarlyRefCount(unsigned lclNum)
+    void UpdateEarlyRefCount(unsigned lclNum, GenTree* node, GenTree* user)
```
Comment (AndyAyersMS) [inline src/coreclr/jit/lclmorph.cpp:1546]: The notes above here are stale, `fgMakeOutgoingStructArgCopy` now relies on early liveness, as does `fgForwardSub` (by default).

We can clean this up later, but if you want to tackle it now there's a bit of dead code and some stale comments over in `fgMakeOutgoingStructArgCopy` too.

**Review by AndyAyersMS — APPROVED** [review]

File: src/coreclr/jit/lclmorph.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/lclmorph.cpp:1544]: Deleted. Looks like substitution also has some stale-ish comments; left those for another day.

**Comment by AndyAyersMS** [comment]
@SingleAccretion, thanks.

**Pull request merged**